
#### ChangeSet ####
2004-09-03 10:27:16-07:00, hirofumi@mail.parknet.co.jp 
  [PATCH] NLS: nls_cp932 fix
  
  Some unicode chars is missing to convert on nls_cp932.
  This patch fixes the following chars.
  
      U+00A2, U+00A3, U+00A7, U+00A8, U+00AC, U+00B0, U+00B1, U+00B4,
      U+00B6, U+00D7, U+00F7, U+FFE2
  
  Signed-off-by: OGAWA Hirofumi <hirofumi@mail.parknet.co.jp>
  Signed-off-by: Andrew Morton <akpm@osdl.org>
  Signed-off-by: Linus Torvalds <torvalds@osdl.org>

==== fs/nls/nls_cp932.c ====
2004-09-03 02:08:22-07:00, hirofumi@mail.parknet.co.jp +39 -4
  NLS: nls_cp932 fix

--- 1.7/fs/nls/nls_cp932.c	2003-09-21 14:49:57 -07:00
@@ -7842,9 +7869,17 @@ static int uni2char(const wchar_t uni,
 		if (out[0] == 0x00 && out[1] == 0x00)
 			return -EINVAL;
 		return 2;
-	} else if ((ch == 0) && (cl <= 0x7F)) {
-		out[0] = cl;
-		return 1;
+	} else if (ch == 0) {
+		if (cl <= 0x7F) {
+			out[0] = cl;
+			return 1;
+		} else if (0xA0 <= cl) {
+			out[0] = u2c_00hi[cl - 0xA0][0];
+			out[1] = u2c_00hi[cl - 0xA0][1];
+			if (out[0] && out[1])
+				return 2;
+		}
+		return -EINVAL;
 	}
 	else
 		return -EINVAL;
